tools/python/api_client/bmapi.py

```diff
@@ -67,27 +43,38 @@ def __init__(self, response_dict):
     if 'BM_SKILL_RAND_VALS_ROLLED' in response_dict:
       self.skill_rand_vals = response_dict['BM_SKILL_RAND_VALS_ROLLED']
 
-class BMClient():
+
+class BMClient:
   def _read_rcfile(self, rcfile, site):
     config = configparser.ConfigParser()
     config.read(rcfile)
     self.url = config.get(site, "url")
     self.username = config.get(site, "username")
     self.password = config.get(site, "password")
-    self.cookiefile = os.path.expanduser(config.get(site, "cookiefile"))
+    try:
+      self.cookiefile = os.path.expanduser(config.get(site, "cookiefile"))
+    except configparser.NoOptionError:
+      self.cookiefile = None
     try:
       self.cachedir = os.path.expanduser(config.get(site, "cachedir"))
-    except (configparser.NoOptionError):
+    except configparser.NoOptionError:
       pass
 
   def _setup_cookies(self):
-    # all requests should use the same cookie jar
-    self.cookiejar = LWPCookieJar(self.cookiefile)
-    if os.path.isfile(self.cookiefile):
-      self.cookiejar.load(ignore_discard=True)
-    self.cookieprocessor = HTTPCookieProcessor(self.cookiejar)
-    self.opener = build_opener(self.cookieprocessor)
-    install_opener(self.opener)
+    # ref session so requests can share session config like a shared cookiejar
+    self.session = requests.session()
+    if self.cookiefile is not None:
+      print('file cookies')
+      self.cookiejar = LWPCookieJar(self.cookiefile)
+      if os.path.isfile(self.cookiefile):
+        print('cookie cache')
+        # load up existing cachefile
+        self.cookiejar.load(ignore_discard=True)
+    else:
+      print('in mem cookies')
+      # use in-memory cookie jar if needed
+      self.cookiejar = CookieJar()
+    self.session.cookies = self.cookiejar
```

Comment:
This seems reasonable, but i think you left some debugging print statements in.

Reply:
hahaha I sure did. thanks for catching this.